framework/console/controllers/MessageController.php

```diff
@@ -429,6 +429,8 @@ protected function extractMessages($fileName, $translator, $ignoreCategories = [
         $this->stdout("Extracting messages from $coloredFileName...\n");
 
         $subject = file_get_contents($fileName);
+        $subject = preg_replace("/<\?[^a-zA-Z0-9\=]*/", '<?php ', $subject);
```

Comment:
This is risk prone: you replaces any appearance of `<?` to `<?php`.
This means if there is a translation message like:
```php
Yii::t('app', 'You can use short tag like "<?"');
```
it will be broken.